packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_splash/blocs/splash_bloc.dart

```diff
@@ -69,15 +73,15 @@ class SplashBloc extends $SplashBloc {
     if (_redirectLocation != null) {
       _navigationBloc.events.goToLocation(_redirectLocation!);
     } else { {{#has_authentication}} {{^enable_pin_code}}
-      await _authService.isAuthenticated()
+      await _authService.isAuthenticated() {{#enable_feature_onboarding}} && !(await _usersService.isProfileTemporary()) {{/enable_feature_onboarding}}
           ? _navigationBloc.events.go(const DashboardRoute())
           : _navigationBloc.events.go(const LoginRoute());
       {{/enable_pin_code}}{{/has_authentication}}
       {{^has_authentication}}
       _navigationBloc.events.go(const DashboardRoute());{{/has_authentication}}
 
       {{#enable_pin_code}}
-      if (await _authService.isAuthenticated()) {
+      if (await _authService.isAuthenticated() {{#enable_feature_onboarding}} && !(await _usersService.isProfileTemporary()) {{/enable_feature_onboarding}}) {
         if(await _pinCodeService.isPinCodeInSecureStorage()) {
           return _navigationBloc.events.go(const VerifyPinCodeRoute(),
               extra: const PinCodeArguments(title: 'Enter Pin Code'));
```

Comment:
We can also localise this as well, can't we?

Reply:
Not easily, since the BuildContext isn't accessible inside the BLoC.

However I agree that this isn't the correct way, but it's outside the scope of this task/PR. We can address it after the Onboarding is done